tests/ipfwd/test_nhop_group.py

```diff
@@ -414,7 +417,7 @@ def test_nhop_group_member_count(request, duthost, tbinfo):
 
     # verify the test used up all the NHOP group resources
     # skip this check on Mellanox as ASIC resources are shared
-    if not is_mellanox_device(duthost):
+    if not is_mellanox_device(duthost) and not is_cisco_device(duthost):
```

Comment:
from Mike Otto,

For next hop group entries available, Gibraltar ASIC has 2 resources that can be used, STAGE1_LB_GROUP and STAGE2_LB_GROUP entries (each having 16K entries). Based on how the NHG is constructed/used will determine which resource it uses, STAGE1 or STAGE2.

We also have 2 resources for NHG MEMBERS, STAGE1_LB_MEMBER and STAGE2_LB_MEMBER where STAGE1 has 192K entries and STAGE2 has 32K entries. When the NHG is constructed using STAGE1_LB_GROUP resource, the members use STAGE1_LB_MEMBERS. When The NHG is constructed using STAGE2_LB_GROUP resource, then the members use STAGE2_LB_MEMBER entries.

When we added support for the IP TUNNEL Path collapse to allow support for dynamic updates to a tunnel destination with better routes, this required creating the NHG using STAGE2_LB_GROUP entries and the members using STAGE2_LB_MEMBER entries. 

So for the NHG CRM test case, as it creates 16K NHGs using STAGE2_LB_GROUP entries and then create 10 members per NHG using STAGE2_LB_MEMBER entries, we would run out (INSUFFIENT RESOURCES), 16K*10=160K > 32K available STAGE2_LB_MEMBER entries.
So, we need to modify the test to keep the number of members under 32K to avoid running out of STAGE2_LB_MEMBER entries.
 
Current SDK drops return 32K (STAGE1_LB_GROUP + STAGE2_LB_GROUP) for the NHG entries instead of 16K. This will be reverted back to only return 16K for NHG available entries as it was prior to adding the IP TUNNEL PATH COLLAPSE support. But this was not the cause of running out of resources on the sonic-mgmt ipfwd/test_nhop_group.py test case. It was due to now using STAGE2 for NHG and NHG Member entries.

Reply:
Can I understand the SDK1.5 is using STAGE1_LB_GROUP/MEMBER to allocate the resource, it can support 16K groups/192K members. But after adding the feature IP TUNNEL PATH COLLAPSE, it shifted to STAGE2_LB_GROUP/MEMBER and the value came to 16K groups/32K members.  
Back to the test case, so the hardware limited the group to 16K, but the CRM will return 32K groups(STAGE1+STAGE2), and you want to make 2 members for each group, that means 64K members is required, but there are just 32K members in STAGE2, looks like it is not enough?

Reply:
SAI/Mike still works on reverting with SDK folks. will keep an update once any result.